src/build/schema/serializer.ts

```diff
@@ -0,0 +1,331 @@
+		type.isInterface() || type.isClass() ||
+		Node.isTypeLiteral(symbolDeclartion)
+	) {
+		const propertiesMap = type.getProperties().map((
+			prop,
+		) => {
+			const type = prop.getValueDeclaration()?.getType() || prop.getTypeAtLocation(prop.getDeclarations()[0]!);
+			return [
+				prop.getName(),
+				generateSerializableTypeSchema(type, { isOptional: prop.isOptional() }),
+			];
+		});
+
+		return s.object(Object.fromEntries(propertiesMap));
+	}
+
+	if (type.isObject()) {
+		const stringIndexType = type.getStringIndexType();
+		if (stringIndexType) {
+			const indexType = generateSerializableTypeSchema(stringIndexType);
+			return s.record(indexType);
+		}
+		const numberIndexType = type.getNumberIndexType();
+		if (numberIndexType) {
+			const indexType = generateSerializableTypeSchema(numberIndexType);
+			return s.record(indexType);
+		}
+		const properties = type.getProperties();
+		const propMap = properties.map((prop) => {
+			const type = prop.getValueDeclaration()?.getType() || prop.getTypeAtLocation(prop.getDeclarations()[0]!);
+			return [
+				prop.getName(),
+				type ? generateSerializableTypeSchema(type) : s.unknown(),
+			];
+		});
+		return s.object(Object.fromEntries(propMap));
+	}
+
+	return s.unknown();
+}
+
+const DEFAULT_COMPILER_OPTIONS = {
+	allowJs: true,
+	esModuleInterop: true,
+	experimentalDecorators: false,
+	isolatedModules: true,
+	strict: true,
+	useDefineForClassFields: true,
+	noLib: true,
+	allowImportingTsExtensions: true,
+};
+
+function generateSerializableDeclarationSchema(
+	declaration:
+		| InterfaceDeclaration
+		| TypeAliasDeclaration
+		| ClassDeclaration,
+): AnySchemaElement {
+	if (
+		Node.isInterfaceDeclaration(declaration) ||
+		Node.isClassDeclaration(declaration)
+	) {
+		const heriatageClauses = declaration.getHeritageClauses();
+		const heriatageClausesSchemas = heriatageClauses.map((clause) => {
+			return clause.getTypeNodes().map((typeNode) => {
+				return generateSerializableTypeSchema(typeNode.getType()!);
+			});
+		}).flat();
+
+		const props = declaration.getProperties().map((prop) => {
+			return [
+				prop.getName(),
+				generateSerializableTypeSchema(prop.getType()!, {
+					isOptional: prop.hasQuestionToken(),
+				}),
+			];
+		});
+
+		return s.object({
+			...Object.fromEntries(props),
+			...heriatageClausesSchemas.reduce((acc, schema) => {
+				if (is("object", schema)) {
+					return {
+						...acc,
+						...schema.properties,
+					};
+				}
+				return acc;
+			}, {}),
+		});
+	}
+	if (Node.isTypeAliasDeclaration(declaration)) {
+		const type = declaration.getType();
+		return generateSerializableTypeSchema(type);
+	}
+
+	return s.unknown();
+}
+
+// This is a stripped down version of the typescript lib
+// that only contains the types we want to serialize.
+// This is to avoid serializing the entire typescript lib which
+// is huge and contains a lot of types we don't need.
+const tsLib = Deno.readTextFileSync(import.meta.dirname + "/schema_ts_lib.ts");
```

Comment:
This likely won't compile in CI, you'll have to add this file to the generated build artifacts and import it from there